pkg/runtime/runtime.go

```diff
@@ -303,9 +308,45 @@ func (a *DaprRuntime) initDirectMessaging(resolver servicediscovery.Resolver) {
 		resolver)
 }
 
-// OnComponentUpdated updates the Dapr runtime with new or changed components
-// This method is invoked from the Dapr Control Plane whenever a component update happens
-func (a *DaprRuntime) OnComponentUpdated(component components_v1alpha1.Component) {
+func (a *DaprRuntime) beginComponentsUpdates() error {
+	if a.runtimeConfig.Mode != modes.KubernetesMode {
+		return nil
+	}
+
+	go func() {
+		client, conn, err := client.GetOperatorClient(a.runtimeConfig.Kubernetes.ControlPlaneAddress)
+		if err != nil {
+			log.Errorf("error connecting to operator: %s", err)
+			return
+		}
+		defer conn.Close()
+
+		stream, err := client.ComponentUpdate(context.Background(), &empty.Empty{})
+		if err != nil {
+			log.Errorf("error from operator stream: %s", err)
+			return
+		}
+		for {
+			c, err := stream.Recv()
+			if err != nil {
+				log.Errorf("error from operator stream: %s", err)
```

Comment:
any error will return from the routine thus stopping any future updates, should it  log and continue,

Reply:
In the case of an error from `stream`, it means the connection has stopped and failed. We will need to retry and renew the stream, but as we are currently not doing anything with the updates, I wouldn't touch this now until further notice.